server/src/main/scala/manifests/graph/BatchRequestsStage.scala

```diff
@@ -0,0 +1,118 @@
+package manifests.graph
+
+import akka.stream.{Attributes, FlowShape, Inlet, Outlet}
+import akka.stream.stage.{GraphStage, GraphStageLogic, InHandler, OutHandler}
+import drt.shared.CrunchApi.MillisSinceEpoch
+import drt.shared.{Arrival, SDateLike}
+import org.slf4j.{Logger, LoggerFactory}
+
+class BatchRequestsStage(now: () => SDateLike) extends GraphStage[FlowShape[List[Arrival], List[SimpleArrival]]] {
+  val inArrivals: Inlet[List[Arrival]] = Inlet[List[Arrival]]("inArrivals.in")
+  val outArrivals: Outlet[List[SimpleArrival]] = Outlet[List[SimpleArrival]]("outArrivals.out")
+
+  override def shape = new FlowShape(inArrivals, outArrivals)
+
+  val log: Logger = LoggerFactory.getLogger(getClass)
+
+  override def createLogic(inheritedAttributes: Attributes): GraphStageLogic = new GraphStageLogic(shape) {
+    var subscribers: Map[SimpleArrival, Option[Long]] = Map()
+    var prioritised: List[SimpleArrival] = List()
+
+    setHandler(inArrivals, new InHandler {
+      override def onPush(): Unit = {
+        val incoming = grab(inArrivals)
+
+        log.info(s"grabbed ${incoming.length} requests for arrival manifests")
+
+        incoming.foreach { arrival =>
+          val simpleArrival = SimpleArrival(arrival)
+          subscribers.get(simpleArrival) match {
+            case None =>
+              addToPrioritised(simpleArrival)
+              addToSubscribers(simpleArrival)
+            case Some(None) if isWithinHours(arrival.Scheduled, 48) => addToPrioritised(simpleArrival)
```

Comment:
we should check if this could result in a flight being added to the prioritised list more than once, perhaps we could write a test for that.

Reply:
Removed this case and added a guard to prevent duplicate lookup requests